transforms/class.js

```diff
@@ -258,24 +258,34 @@ module.exports = (file, api, options) => {
       })
       .forEach(path => j(path).replaceWith(j.identifier('props')));
 
-  const inlineGetInitialState = getInitialState =>
-    getInitialState.value.body.body.map(statement => {
-      if (statement.type === 'ReturnStatement') {
-        return j.expressionStatement(
+  const inlineGetInitialState = getInitialState => {
+    const functionExpressionAST = j(getInitialState.value);
+
+    return functionExpressionAST
+      .find(j.ReturnStatement)
+      .forEach(path => {
+        j(path).replaceWith(j.expressionStatement(
           j.assignmentExpression(
             '=',
             j.memberExpression(
               j.thisExpression(),
               j.identifier('state'),
               false
             ),
-            statement.argument
+            path.value.argument
           )
-        );
-      }
-
-      return statement;
-    });
+        ));
+
+        if ( // FIXME is there a better way to check this?
```

Comment:
The current implementation assumes that people don't call `createClass` inside a big `if` statement. 😨 Also it doesn't handle stuff like try catch or promise properly. I think a better way is to get the `closestScope` of the `return` and compare it with the scope that `getInitialState` creates -- do you know if there's a way to compare 2 scopes?